### PR TITLE
added extra solution to pig latin guide

### PR DESCRIPTION
**Pull request by rulamon**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #XXXXX

### CODE REVIEW

**Review by RandellDawson — APPROVED** [review]
LGTM.  

**Other Reviewers:** There is an earlier PR (#34446) using regex, that I think should be merged first (if approved) and then consider if we should replace that solution with this one or have them both.    Keep in mind there are already 2 other solutions using regex in some way.

**Review by Manish-Giri — COMMENTED** [review]
@RandellDawson I reviewed the solutions proposed in #34446 and this one, and I feel we should merge the solution in #34446 as "another" intermediate solution, and we should replace the [existing "advanced" solution](https://github.com/freeCodeCamp/freeCodeCamp/blob/71c854c0a13317077b44e9f7751375d7bbab73a0/guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/pig-latin/index.md#-advanced-code-solution) with the one in this PR.

**Comment by RandellDawson** [comment]
@Manish-Giri I agree with your assessment of these two PRs.  Are you going to modify the PRs to put them in the correct sections or ask the contributors to do so?

**Comment by Manish-Giri** [comment]
@RandellDawson I think we can merge #34446 rightaway. After that I can either edit this PR to fix the potential merge conflicts and re-arrange this section, or we can ask the OP to rebase and re-submit the PR, but it might be too much to ask, since they're a first time contributor.

File: guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/pig-latin/index.md [hunk 1]
```diff
@@ -148,56 +146,28 @@ You will need to use everything you know about string manipulation to get the la
...
-        // push all consonats to the end of the array
-        while (isConsonant(strArr[0])) {
-            tmpChar = strArr.shift();
-            strArr.push(tmpChar);
-        }
-     // convert array to string and concatenate "ay" at the end  
-     return strArr.join("")+"ay";
-    }
+      return str.replace(/^[aeiou]\w*/, "$&way").replace(/(^[^aeiou]+)(\w*)/, "$2$1ay");
+      }
```
Comment (RandellDawson) [inline guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/pig-latin/index.md]: Looks like this bracket should be moved back two spaces.

**Comment by RandellDawson** [comment]
LGTM.  I will merge once the build completes.

**Comment by RandellDawson** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 🎉

**Pull request merged**